File: src/services/tauri/mcp.ts
import {
  type UnifiedMcpClientName,
  type UnifiedMcpConfig,
  invokeTauri,
  isTauri,
  postJson,
  postNoContent,
} from './shared';

export async function unifiedReadMcpConfig(
  clientName: UnifiedMcpClientName,
  path?: string
): Promise<UnifiedMcpConfig> {
  if (isTauri()) {
    return await invokeTauri<UnifiedMcpConfig>('unified_read_mcp_config', { clientName, path });
  }
  return await postJson<UnifiedMcpConfig>('/api/codex/mcp/read', { client_name: clientName, path });
}

export async function unifiedAddMcpServer(params: {
  clientName: UnifiedMcpClientName;
  path?: string;
  serverName: string;
  serverConfig: unknown;
  scope?: string;
}) {
  if (isTauri()) {
    await invokeTauri('unified_add_mcp_server', params);
    return;
  }
  await postNoContent('/api/codex/mcp/add', {
    client_name: params.clientName,
    path: params.path,
    server_name: params.serverName,
    server_config: params.serverConfig,
    scope: params.scope,
  });
}

export async function unifiedRemoveMcpServer(params: {
  clientName: UnifiedMcpClientName;
  path?: string;
  serverName: string;
  scope?: string;
}) {
  if (isTauri()) {
    await invokeTauri('unified_remove_mcp_server', params);
    return;
  }
  await postNoContent('/api/codex/mcp/remove', {
    client_name: params.clientName,
    path: params.path,
    server_name: params.serverName,
    scope: params.scope,
  });
}

export async function unifiedEnableMcpServer(params: {
  clientName: UnifiedMcpClientName;
  path?: string;
  serverName: string;
}) {
  if (isTauri()) {
    await invokeTauri('unified_enable_mcp_server', params);
    return;
  }
  await postNoContent('/api/codex/mcp/enable', {
    client_name: params.clientName,
    path: params.path,
    server_name: params.serverName,
  });
}

export async function unifiedDisableMcpServer(params: {
  clientName: UnifiedMcpClientName;
  path?: string;
  serverName: string;
}) {
  if (isTauri()) {
    await invokeTauri('unified_disable_mcp_server', params);
    return;
  }
  await postNoContent('/api/codex/mcp/disable', {
    client_name: params.clientName,
    path: params.path,
    server_name: params.serverName,
  });
}

File: src-tauri/src/commands/mcp.rs
use crate::features::mcp;
use serde_json::Value as JsonValue;

#[tauri::command]
pub async fn unified_add_mcp_server(
    client_name: String,
    path: Option<String>,
    server_name: String,
    server_config: JsonValue,
    scope: Option<String>,
) -> Result<(), String> {
    mcp::unified_add_mcp_server(client_name, path, server_name, server_config, scope).await
}

#[tauri::command]
pub async fn unified_enable_mcp_server(
    client_name: String,
    path: Option<String>,
    server_name: String,
) -> Result<(), String> {
    mcp::unified_enable_mcp_server(client_name, path, server_name).await
}

#[tauri::command]
pub async fn unified_disable_mcp_server(
    client_name: String,
    path: Option<String>,
    server_name: String,
) -> Result<(), String> {
    mcp::unified_disable_mcp_server(client_name, path, server_name).await
}

#[tauri::command]
pub async fn unified_remove_mcp_server(
    client_name: String,
    path: Option<String>,
    server_name: String,
    scope: Option<String>,
) -> Result<(), String> {
    mcp::unified_remove_mcp_server(client_name, path, server_name, scope).await
}

#[tauri::command]
pub async fn unified_read_mcp_config(
    client_name: String,
    path: Option<String>,
) -> Result<JsonValue, String> {
    mcp::unified_read_mcp_config(client_name, path).await
}
